pkg/config/manifest.go

```diff
@@ -559,6 +577,34 @@ func (m *Manifest) SliceElem(index int, val reflect.Value) error {
 			}
 			val.Set(reflect.ValueOf(r))
 		}
+	} else {
+		v := val
+		if val.Kind() == reflect.Interface {
+			val = val.Elem()
+		}
+		if kind := val.Kind(); kind == reflect.Map {
+			if len(val.MapKeys()) == 1 {
+				sk := val.MapKeys()[0]
+				if sk.Kind() == reflect.Interface {
+					sk = sk.Elem()
+				}
+				//if the key is a string, and the string is source, then we should try
```

Comment:
Like if we stop supporting `- "source: bundle...."` does this entire block go away?

Reply:
No, the opposite. This got added to support the other case where there are no quotes. 

This is the part that handles the quotes:

```
v, ok := val.Interface().(string)
	if ok {
		//if the array entry is a string that matches source:...., we should replace it
		re := regexp.MustCompile("source:\\s?(.*)")
		matches := re.FindStringSubmatch(v)
		if len(matches) > 0 {
			source := matches[1]
			r, err := m.resolveValue(source)
			if err != nil {
				return errors.Wrap(err, "unable to source value")
			}
			val.Set(reflect.ValueOf(r))
		}
	}
```

Reply:
Oh noes, I've been found out. I have a really hard time reading this reflection code.

I'm ok merging this as-is. Up to you if you think we can just remove the "source: bundle...." case and make things simpler or not.